packages/flutter_tools/lib/src/commands/packages.dart

```diff
@@ -253,3 +258,82 @@ class PackagesPassthroughCommand extends FlutterCommand {
     return FlutterCommandResult.success();
   }
 }
+
+class PackagesInteractiveGetCommand extends FlutterCommand {
+  PackagesInteractiveGetCommand(this._commandName, this._description) {
+    requiresPubspecYaml();
+  }
+
+  @override
+  ArgParser argParser = ArgParser.allowAnything();
+
+  final String _commandName;
+  final String _description;
+
+  @override
+  String get name => _commandName;
+
+  @override
+  String get description {
+    return '$_description.\n'
+           'This runs the "pub" tool in a Flutter context.';
+  }
+
+  @override
+  String get invocation {
+    return '${runner.executableName} pub $_commandName [<arguments...>]';
+  }
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    List<String> rest = argResults.rest;
+    String target;
+    if (rest.length == 1 &&
+        (rest[0].contains('/') ||
+            rest[0].contains(r'\'))) {
+      // HACK: Supporting flutter specific behavior where you can pass a
+      //       folder to the command.
+      target = findProjectRoot(rest[0]);
+      rest = <String>[];
+    } else {
+      target = findProjectRoot();
+    }
+    if (target == null) {
+      throwToolExit('Expected to find project root in '
+          'current working directory.');
+    }
+    final FlutterProject flutterProject = FlutterProject.fromPath(target);
+
+    if (flutterProject.manifest.generateSyntheticPackage) {
+      final Environment environment = Environment(
+        artifacts: globals.artifacts,
+        logger: globals.logger,
+        cacheDir: globals.cache.getRoot(),
+        engineVersion: globals.flutterVersion.engineRevision,
+        fileSystem: globals.fs,
+        flutterRootDir: globals.fs.directory(Cache.flutterRoot),
+        outputDir: globals.fs.directory(getBuildDirectory()),
+        processManager: globals.processManager,
+        projectDir: flutterProject.directory,
+      );
+
+      await generateLocalizationsSyntheticPackage(
+        environment: environment,
+        buildSystem: globals.buildSystem,
+      );
+    }
+
+    final List<String> subArgs = rest.toList()
+      ..removeWhere((String arg) => arg == '--');
+    await pub.interactively(
```

Comment:
@jonahwilliams Can this class be tested by mocking out `pub` here? E.g. a test could make sure the argument manipulation logic above is correct.